libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -939,7 +964,6 @@ void BeaconSetupService::OnDkgMessage(MuddleAddress const &              from,
     auto ptr = std::dynamic_pointer_cast<FinalStateMessage>(msg_ptr);
     if (ptr != nullptr)
     {
-      // TODO(HUT): might be unnecessary to check given guarantee of reliable broadcast
       if (final_state_payload_.find(from) == final_state_payload_.end())
```

Comment:
Not true, the RBC does prevent a node from broadcasting multiple messages of the same type.